### DISCUSSION TITLE
Possibility to avoid node/npm/yarn with Stimulus?

### DISCUSSION THREAD

**Discussion by yerlaser** [ruby2js/ruby2js#143] [Q&A] [answered]
Since Stimulus doesn't require a build step and can work with just module import, would it be possible to somehow avoid node/npm/yarn setup to use it?

**Comment by rubys** [comment]
@yerlaser can you explain a bit more what you are looking for?  Some sort of build step is required as browsers don't natively support Ruby.  It is possible to do the conversion without node, there is code in `demo/ruby2js.rb` to do so.

Snowpack can do this build step transparently on demand.

With Rails, it once was possible to have sprockets do the build - it was working with Rails 6; I haven't checked to see if it could work with Rails 7.

**Reply by yerlaser** [reply]
Hi,
Yes, I basically want ruby2j to only compile rb to js and then load that semi-manually in an html file.
The docs I've read so far gave me an impression that nodejs was somehow required.
Perhaps, I missed something.
I'll look at more demos then, thanks.

**Comment by rubys** [comment, marked as answer]
The [README](https://github.com/ruby2js/ruby2js/blob/master/README.md#synopsis) gives an overview of how ruby2js can be called as an API.  I've now committed a [change](https://github.com/ruby2js/ruby2js/commit/cc39ce46f6210532021f8c72416891b6edda69a1) that will install the demo as a command added to your PATH in future versions of the ruby2js gem.

**Reply by yerlaser** [reply]
That is great. Thank you!
